sklearn/decomposition/_nmf.py

```diff
@@ -248,6 +248,22 @@ def _beta_loss_to_float(beta_loss):
     return beta_loss
 
 
+def _check_params(n_components, max_iter, tol):
```

Comment:
This function should be a method of the class. Same remark for _check_w_h

Reply:
Knowing that MiniBatchNMF will follow why not leaving _check_w_h and _check_params outside the class as they could be reused? This is the same for eg _check_init, am I wrong?

Reply:
MiniBatchNMF will inherit from NMF so it will have access to the these methods

Reply:
I think that _check_init and some others should also be methods of the class but I didn't say it to not make the diff too big

Reply:
I find very confusing as a non-expert reader the idea of putting some functions in the class and some other not: I can't see the reason to make a difference. I'm happy to learn if there is one, but could we put all the functions in the same place if not? Thanks.

Reply:
To me the functions meant to validate class parameters should definitely be methods. They're not and never will be used outside of the class. For the functions that "do the math" I don't think it's mandatory. We can imagine them being used somewhere else, and leaving them as functions makes them a bit easier to unit test.

I'd rather have the new functions be put in the right place right away. The other ones can be moved in this PR or later. For my part I find it more confusing to introduce new code in a place I think it should not be.